app/controllers/tax_form_due_dates_controller.rb

```diff
@@ -1,13 +1,28 @@
 class TaxFormDueDatesController < ApplicationController
   before_action :set_tax_form_due_date, only: [:show]
 
-  # GET /tax_form_due_dates
+  # GET /tax_form_due_dates -- params entity_type and jurisdiction -TBD
   def index
-    @tax_form_due_dates = TaxFormDueDate.all
+    search = params[:searchkey]
+    entity_type = params[:type]
+    jurisdiction = params[:jurisdiction]
+    if params[:searchkey] || params[:type] || params[:jurisdiction]
+      @tax_form_due_dates = TaxFormDueDate.datasearch(search, entity_type, jurisdiction).order(:jurisdiction)
+    end
 
     render json: @tax_form_due_dates
   end
 
+
+  # *****************************************************************
+  # to be used in filter options
```

Comment:
there is a better way of getting these options. also, this should not be in the controller logic